### PR TITLE
[App Configuration] Update snapshot API

### PR DESCRIPTION
**Pull request by ShivangiReja**
Here's the following changes in the PR:

* Removed `SnapshotName` property from SettingSelector type
* Added GetConfigurationSettingsForSnapshot method and related tests
* Updated TSG

### CODE REVIEW

File: sdk/appconfiguration/Azure.Data.AppConfiguration/TROUBLESHOOTING.md [hunk 1]
```diff
@@ -109,6 +110,19 @@ ConfigurationClientOptions options = new ConfigurationClientOptions()
 ```
 
 For more detail on capturing Azure SDK logs and additional examples, please see the [Azure SDK diagnostics sample](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md).
+
+### Limit Issues
+
+The error thrown by the App Configuration client library includes a detailed response error object that provides specific useful insights into what went wrong and includes corrective actions to fix common issues. A common error encountered is HTTP status code 429 for exceeding the limit. Refer to the body of the 429 response for the specific reason why the request failed. The failures often happen under these circumstances:
```
Comment (jsquire) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/TROUBLESHOOTING.md:116]: Do we have any docs that we could link to that describe the rate limit?

**Review by jsquire — APPROVED** [review]

File: sdk/appconfiguration/Azure.Data.AppConfiguration/TROUBLESHOOTING.md (hunk 1 above)
Comment (ShivangiReja) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/TROUBLESHOOTING.md:116]: I found these 2 docs:
* https://learn.microsoft.com/en-us/azure/azure-app-configuration/faq#are-there-any-limits-on-the-number-of-requests-made-to-app-configuration
* https://azure.microsoft.com/en-us/updates/general-availability-azure-app-configuration-now-offers-increased-hourly-request-limit/

Which one do you think we should link here? I think FAQ one is linked with AppConfig docs, so we can use it in our TSG.

Comment (jsquire) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/TROUBLESHOOTING.md]: oohh... the first one!

Comment (ShivangiReja) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/TROUBLESHOOTING.md]: Updated!

**Pull request merged**